src/ui/session/content/chat_action_bar.rs

```diff
@@ -135,13 +135,19 @@ mod imp {
                 |widget, _, _| async move {
                     let chat = widget.chat().unwrap();
                     if let model::ChatType::Private(user) = chat.chat_type() {
+                        let is_blocked =
+                            if let tdlib::enums::BlockList::Main = chat.block_list().unwrap().0 {
+                                None
+                            } else {
+                                Some(tdlib::enums::BlockList::Main)
+                            };
                         let message_sender =
                             tdlib::enums::MessageSender::User(tdlib::types::MessageSenderUser {
                                 user_id: user.id(),
                             });
-                        let result = tdlib::functions::toggle_message_sender_is_blocked(
+                        let result = tdlib::functions::set_message_sender_block_list(
                             message_sender,
-                            !chat.is_blocked(),
+                            is_blocked,
```

Comment:
I think it's better to just pass `None` here, since the action is called `unblock-chat` and not `toggle-block-chat`. You only inherited this, to be fair.